core/src/epicli/data/common/ansible/playbooks/roles/preflight/tasks/apply.yml

```
@@ -61,6 +63,13 @@
           {{ (_defined | selectattr('config.enabled') | list) + _undefined }}
         _unsupported_enabled_extensions: >-
           {{ _enabled_extensions | selectattr('name', 'in', unsupported_postgres_extensions[_pg_arch]) | list }}
+
+    - name: Check the number of PostgreSQL hosts for Repmgr
+      when: postgresql.specification.extensions.replication.enabled | default(false)
+      assert:
+        that: groups.postgresql | length <= 2
```

```suggestion
        that: groups.postgresql | length == 2
```
How about requiring 2 hosts when replication is enabled?

---

Seems like hard restriction to user, as repmgr could be installed without a second node. Now I don't see any case to do so, but who knows what user wants?

---

changed by @to-bar